### PR TITLE
cleanup tck dist build

### PR DESCRIPTION
**Pull request by lukasj**
Signed-off-by: Lukas Jungmann <lukas.jungmann@oracle.com>

### CODE REVIEW

File: tck/pom.xml [hunk 1]
```diff
@@ -62,44 +63,90 @@
...
+            <dependency>
+                <groupId>jakarta.json</groupId>
+                <artifactId>jakarta.json-tck-ug</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>jakarta.json</groupId>
+                <artifactId>jakarta.json-api</artifactId>
+                <version>2.1.0-SNAPSHOT</version>
```
Comment (jansupol) [inline tck/pom.xml:82]: `${project.version}` ?

File: api/pom.xml [hunk 2]
```diff
@@ -77,7 +77,7 @@
     <properties>
         <copyright.exclude>${project.basedir}/../etc/copyright-exclude</copyright.exclude>
         <copyright.templatefile>${project.basedir}/../etc/copyright.txt</copyright.templatefile>
-        <copyright.ignoreyear>false</copyright.ignoreyear>
+        <copyright.ignoreyear>true</copyright.ignoreyear>
```
Comment (jansupol) [inline api/pom.xml:80]: Copyright year update to 2022

File: tck/tck-dist/src/main/assembly/assembly.xml [hunk 3]
```diff
@@ -19,68 +19,43 @@
...
-                <include>*.*</include>
-            </includes>
-         </fileSet>
-         <fileSet>
-            <directory>../tck-docs/userguide/target/generated-docs</directory>
-            <outputDirectory>docs/pdf-usersguide</outputDirectory>
+    <id>distribution</id>
+    <baseDirectory>jsonp-tck</baseDirectory>
+    <formats>
+        <format>zip</format>
```
Comment (jansupol) [inline tck/tck-dist/src/main/assembly/assembly.xml:25]: Fyi: TCK in EE10 is acceptable in a form of jar, too, AFAIK.

File: tck/pom.xml (hunk 1 above)
Comment (lukasj) [reply tck/pom.xml:82]: not necessarily - tck and api version may be different/have different lifecycle - ie does minor update of the spec jar require and update of TCK?

File: api/pom.xml (hunk 2 above)
Comment (lukasj) [reply api/pom.xml:80]: can you point me to the file modified in this PR which does not have updated cp year?

File: tck/tck-dist/src/main/assembly/assembly.xml (hunk 3 above)
Comment (lukasj) [reply tck/tck-dist/src/main/assembly/assembly.xml:25]: https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/ shows only zip files

Comment (jansupol) [reply tck/tck-dist/src/main/assembly/assembly.xml:25]: Right, I guess Jakarta REST would be the first one...

File: api/pom.xml (hunk 2 above)
Comment (jansupol) [reply api/pom.xml:80]: This very file api/pom.xml

File: tck/pom.xml (hunk 1 above)
Comment (jansupol) [reply tck/pom.xml:82]: True, it can have a different lifecycle. Never mind how confusing that would be.

**Comment by Manuel13dpl** [comment]
Ola espero me pueda. Ayudar a abrir todos mis archivos ya q me aparece q está protegido por pat joy 
- -@jsonp-bot [](te lo agradeceria mucho

**Comment by lukasj** [comment]
I also went ahead and changed project artifact IDs in the following way:

`jakarta.json:jakarta.json-tck-project` - parent for tck related projects (was `jakarta.json:jakarta.json-tck`)
`jakarta.json:jakarta.json-tck-common` - untouched
`jakarta.json:jakarta.json-tck-tests` - untouched
`jakarta.json:jakarta.json-tck-tests-pluggability` - fixed typo in the name (was `jakarta.json:jakarta.json-tck-tests-plugability`)
`jakarta.json:jakarta.json-tck` - the distribution zip (was `jakarta.json:jsonp-tck-dist`)

File: tck/tck-dist/src/main/assembly/assembly.xml (hunk 3 above)
Comment (lukasj) [reply tck/tck-dist/src/main/assembly/assembly.xml:25]: my thinking is that a `.jar` is something I can put on the classpath and/or run directly which is not true with this particular artifact - at least not yet. We could certainly provide a `.jar` containing tests only (say current `jakarta.json:jakarta.json-tck-tests` artifact) and a `.zip` distribution for offline use (with the userguide etc)

File: api/pom.xml (hunk 2 above)
Comment (lukasj) [reply api/pom.xml:80]: ah, right, fixed

**Review by gurunrao — APPROVED** [review]

**Review by jansupol — APPROVED** [review]

**Pull request merged**